src/mcts/node.cc

```diff
@@ -113,9 +113,50 @@ Move Edge::GetMove(bool as_opponent) const {
   return m;
 }
 
+// We use strictly 16 bits to store policy values in memory-critical Edges.
+// We manage this by storing 16 bits taken directly from the float.
+// Floats are 1 sign bit, 8 bits of exponent, and 23 significand bits. The
+// exponent is stored a bit funny, in such a way that its most significant bit
+// is always off in the interval [0, 2). Furthermore, the next two bits are only
+// off for microscopic floats near 0; we ignore those and round up to the
+// smallest possible 5-bit-exp float above 0 (i.e. assume those 2 bits are set).
+//
+// See https://www.h-schmidt.net/FloatConverter/IEEE754.html for details.
+//
+// Note that that means we don't actually get 2^16 unique values in [0,1],
+// rather some spread into (1,2), but (although it's possible) it's more
+// complicated to make use of the extra values. At any rate, it's true that the
+// available values are disproportionately clustered (and thus more precise)
+// towards 0, which is how NN policy behaves anyways (since policy always adds
+// up to 1). We also implement round-to-nearest by shifting the significand up
+// by half the precision that is ultimately kept (addition overflow into the 7
+// exponent bits is possible, but also works correctly).
+
+// Toss the sign and 3 highest bits of the exponent (assuming them to be 011),
+// take the next 16 bits, that leaves the 12 trailing LSB of the significand to
+// be lost as well. Round-to-nearest by adding a 1 in that 12th bit before
+// truncating. The minimum number that can be represented is 4.656613E-10.
+void Edge::SetP(float p) {
+  assert(0.0f <= p && p <= 1.0f);
+  int32_t tmp;
+  std::memcpy(&tmp, &p, sizeof(float));
+  // If those 2 exponent bits (3<<28) aren't set, round up by setting tmp = 0.
+  // This is combined with the round-to-nearest addition (1<<11) into one op.
+  tmp -= 0x2FFFF800; // == (3 << 28) - (1 << 11)
```

Comment:
If `(3 << 28) - (1 << 11)` is more readable, maybe write it directly as an expression rather than comment?

Reply:
Done, but still not entirely sure about the proposed solution. Please re-review. I want to emphasize that they're conceptually different, even though they can be implemented as one unified cpu instruction.

Reply:
Not really "Done", but ok. :)

Reply:
Ah, actually done! It just showed previous version.